src/Translation.jl

```diff
@@ -0,0 +1,338 @@
+false
+```
+
+The list of constraints of the translation of a polyhedron (in general, a set
+whose `constraints_list` is available) can be computed from a lazy translation:
+
+```jldoctest translation
+julia> constraints_list(tr)
+6-element Array{HalfSpace{Float64},1}:
+ HalfSpace{Float64}([1.0, 0.0, 0.0], 5.0)
+ HalfSpace{Float64}([0.0, 1.0, 0.0], 3.0)
+ HalfSpace{Float64}([0.0, 0.0, 1.0], 3.0)
+ HalfSpace{Float64}([-1.0, -0.0, -0.0], -3.0)
+ HalfSpace{Float64}([-0.0, -1.0, -0.0], -1.0)
+ HalfSpace{Float64}([-0.0, -0.0, -1.0], -1.0)
+```
+"""
+struct Translation{N<:Real, VN<:AbstractVector{N}, S<:LazySet{N}} <: LazySet{N}
+    X::S
+    v::VN
+
+    # default constructor with dimension check
+    function Translation(X::S, v::VN) where {N, VN<:AbstractVector{N}, S<:LazySet{N}}
+        @assert dim(X) == length(v) "cannot create a translation of a set of dimension $(dim(X)) " *
+                                    " along a vector of length $(length(v))" 
+        return new{N, VN, S}(X, v)
+    end
+end
+
+@static if VERSION < v"0.7-"
+    @eval begin
+        # convenience constructor without type parameter
+        Translation(X::S, v::VN) where {N<:Real, VN<:AbstractVector{N}, S<:LazySet{N}} =
+            Translation{N, VN, S}(X, v)
+    end
+end
+
+# constructor from a Translatio:: perform the translation immediately
+Translation(tr::Translation{N}, v::AbstractVector{N}) where {N<:Real} =
+    Translation(tr.X, tr.v + v)
+
+"""
+    X + v
+
+Convenience constructor for a translation.
+
+### Input
+
+- `X` -- convex set
+- `v` -- vector
+
+### Output
+
+The symbolic translation of ``X`` along vector ``v``.
+"""
++(X::LazySet, v::AbstractVector) = Translation(X, v)
+
+"""
+    ⊕(X::LazySet, v::AbstractVector)
+
+Unicode alias constructor ⊕ (`oplus`) for the lazy translation operator.
+"""
+⊕(X::LazySet, v::AbstractVector) = Translation(X, v)
+
+# ============================ 
+# LazySet interface functions
+# ============================
+
+"""
+    dim(tr::Translation)::Int
+
+Return the dimension of a translation.
+
+### Input
+
+- `tr` -- translation
+
+### Output
+
+The dimension of a translation.
+"""
+function dim(tr::Translation)::Int
+    return length(tr.v)
+end
+
+
+"""
+    σ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+
+Return the support vector of a translation.
+
+### Input
+
+- `d`  -- direction
+- `tr` -- translation
+
+### Output
+
+The support vector in the given direction.
+If the direction has norm zero, the result depends on the wrapped set.
+"""
+function σ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+    return tr.v + σ(d, tr.X)
+end
+
+"""
+    ρ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+
+Return the support function of a translation.
+
+### Input
+
+- `d`  -- direction
+- `tr` -- translation
+
+### Output
+
+The support function in the given direction.
+"""
+function ρ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+    return ρ(d, Singleton(tr.v)) + ρ(d, tr.X)
```

Comment:
While I like the readability, we could have inline the definition of `ρ(d, Singleton(tr.v))` to avoid the creation of the object.

Reply:
👍  which is `dot(d, tr.v)`